homeassistant/helpers/intent.py

```diff
@@ -110,51 +114,117 @@ class IntentUnexpectedError(IntentError):
     """Unexpected error while handling intent."""
 
 
-@callback
-@bind_hass
-def async_match_state(
-    hass: HomeAssistant, name: str, states: Iterable[State] | None = None
-) -> State:
-    """Find a state that matches the name."""
-    if states is None:
-        states = hass.states.async_all()
+def _is_device_class(
+    state: State,
+    entity: entity_registry.RegistryEntry | None,
+    device_classes: Collection[str],
+) -> bool:
+    """Return true if entity device class matches."""
+    # Try entity first
+    if (entity is not None) and (entity.device_class is not None):
+        # Entity device class can be None or blank as "unset"
+        if entity.device_class in device_classes:
```

Comment:
I am not sure if we can already assume that all device classes are using the enums

Reply:
Should this be case folded then?